### PR TITLE
games-server/minecraft-server: New package

### PR DESCRIPTION
**Pull request by ConiKost**
games-server/minecraft-server: New package

Closes: https://bugs.gentoo.org/652338

### CODE REVIEW

**Comment by Dessa** [comment]
@chewi this is part of https://github.com/gentoo/java-overlay, maybe you want to look at it?

**Comment by chewi** [comment]
I suppose being the Java lead, the acting Games lead, familiar with Minecraft, and the author of older Minecraft packages means I have to take this ticket. :stuck_out_tongue:

We do already have a package for this in java-overlay. It is 5 years old but it was quite heavily used at the time. I did attempt to bump it 3 years ago but became tied up with the last remaining bundled dependencies. I have no idea what the situation is with bundled libraries now but I would still like to unbundle even some of them if possible. I need to get my urgent work on Portage done before I can take a look at this though.

**Comment by ConiKost** [comment]
@chewi: I am sorry, but I wasn't aware, that the minecraft-server is already part of the java-overlay. It was not my intention to disturb your work :) Looking at those eBuilds in the overlay, I can see, I was using them years ago too. This here is my eBuild, which I am using now for about 2 years in my local overlay.

> I have no idea what the situation is with bundled libraries now but I would still like to unbundle even some of them if possible.

What do you mean by unbundle libraries? Are we talking about the server? That's only a *.jar, which is started by Java. As far I know, no special libraries are needed? Or do you mean the client?

**Comment by jonasstein** [comment]
May/Do we really want to support this? We may not mirror and the package will need a lot of care. It is even restrict bindist.
From the not so nice EULA:

```
+   The one major rule is that you must not distribute anything we've made
+   unless we specifically agree to it. By "distribute anything we've made"
+   what we mean is:
+
+     • give copies of our Game to anyone else;
+     • make commercial use of anything we've made;
+     • try to make money from anything we've made; or
+     • let other people get access to anything we've made in a way that is
+       unfair or unreasonable;
```

**Comment by chewi** [comment]
> What do you mean by unbundle libraries? Are we talking about the server? That's only a *.jar, which is started by Java. As far I know, no special libraries are needed? Or do you mean the client, which uses bundled libraries, as far I know?

I haven't checked the latest but run `jar tf` against the latest and you'll probably see what I mean.

> From the not so nice EULA:

This was present even back when I worked on it. I didn't add bindist but I probably should have. Still, I think that would cover it. It seems to apply more to the game itself (which costs money) than the server (which doesn't). Also a lot has changed since that was written, not least that Mojang was bought by Microsoft. Not that it makes things any better. :wink:

**Comment by ConiKost** [comment]
> I haven't checked the latest but run jar tf against the latest and you'll probably see what I mean.

I see, what you mean. But the question would be, is this the effort worth for unbundle this?

File: games-server/minecraft-server/minecraft-server-1.12.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+S="${WORKDIR}"
+
+pkg_setup() {
+	enewgroup minecraft
+	enewuser minecraft -1 -1 /var/lib/minecraft-server minecraft
+}
+
+src_unpack() {
+	# Don't unpack that jar, just copy it to WORKDIR
+	cp "${DISTDIR}"/"${A}" "${WORKDIR}"
```
Comment (mgorny) [inline games-server/minecraft-server/minecraft-server-1.12.2.ebuild]: ||die

File: games-server/minecraft-server/minecraft-server-1.12.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	insinto /usr/share/games/minecraft-server
+	newins minecraft-server-${PV}.jar minecraft-server.jar
+
+	newinitd "${FILESDIR}"/minecraft-server.initd minecraft-server
+	newconfd "${FILESDIR}"/minecraft-server.confd minecraft-server
+
+	keepdir /var/lib/minecraft-server
+	keepdir /var/log/minecraft-server
+
+	fowners -R minecraft:minecraft /usr/share/games/minecraft-server /var/lib/minecraft-server /var/log/minecraft-server
```
Comment (mgorny) [inline games-server/minecraft-server/minecraft-server-1.12.2.ebuild]: `diropts`, `insopts`?

File: games-server/minecraft-server/minecraft-server-1.12.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+pkg_postinst() {
+	elog "This package provides an init script and a conf file."
```
Comment (mgorny) [inline games-server/minecraft-server/minecraft-server-1.12.2.ebuild]: Please use more regular text wrapping, at ~72 output chars. Your weird wrapping causes a really ragged output.

File: games-server/minecraft-server/minecraft-server-1.12.2.ebuild (hunk 1 above)
Comment (ConiKost) [reply games-server/minecraft-server/minecraft-server-1.12.2.ebuild]: Thanks, fixed.

File: games-server/minecraft-server/minecraft-server-1.12.2.ebuild (hunk 2 above)
Comment (ConiKost) [reply games-server/minecraft-server/minecraft-server-1.12.2.ebuild]: Thanks, changed.

File: games-server/minecraft-server/minecraft-server-1.12.2.ebuild (hunk 3 above)
Comment (ConiKost) [reply games-server/minecraft-server/minecraft-server-1.12.2.ebuild]: Done.

File: games-server/minecraft-server/files/minecraft-server.initd [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	checkpath -f -o "${command_user}:${command_group}" -q "${minecraft_path}"/eula.txt
+	echo "eula=true" > "${minecraft_path}"/eula.txt
+
+	checkpath -d -o "${command_user}:${command_group}" -q "${minecraft_logs}"/"${instance}"
+
+	if [ ! -L "${minecraft_path}"/logs ]; then
+		cd "$minecraft_path}" && ln -s ../../../log/minecraft-server/"${instance}" logs
+	fi
+
+	if [[ -z ${MINECRAFT_OPTS} ]] ; then
```
Comment (mgorny) [inline games-server/minecraft-server/files/minecraft-server.initd]: Bashism.

Comment (ConiKost) [reply games-server/minecraft-server/files/minecraft-server.initd]: Changed.

**Comment by mgorny** [comment]
@chewi, could you take a final look?

**Comment by ConiKost** [comment]
Rebased!

**Pull request closed without merging**